.github/workflows/reusable-docker-workflow.yml

```diff
@@ -0,0 +1,55 @@
+name: Reusable Docker Workflow
+
+on:
+  workflow_call:
+    inputs:
+      # Target
+      repository:
+        required: true
+        type: string
+      version-ref:
+        required: true
+        type: string
+      
+      # Build
+      build-cmd:
+        required: true
+        type: string
+      
+      # Reference image
+      reference-image:
+        required: true
+        type: string
+
+jobs:
+  build:
+    runs-on: ubuntu-24.04
+    steps:
+      # Clone repository whose reproducibility is to be tested
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          persist-credentials: false
+          ref: ${{ inputs.version-ref }}
+          repository: ${{ inputs.repository }}
+
+      # Install diffoci tool to help diff images
+      - name: Install diffoci
+        run: |
+          DIFFOCI_VERSION=v0.1.6
+          curl -L https://github.com/reproducible-containers/diffoci/releases/download/${DIFFOCI_VERSION}/diffoci-${DIFFOCI_VERSION}.linux-amd64 -o /usr/local/bin/diffoci
+          chmod +x /usr/local/bin/diffoci
```

Comment:
Would've like to check [attestations](https://github.com/reproducible-containers/diffoci/attestations) here but unfortunately it doesn't look like `diffoci` has them 😢

Could use the [GitHub CLI](https://cli.github.com/) here instead of `curl` which might be a bit more expressive 🤔 up to you.